confmap/confmap.go

```diff
@@ -287,7 +288,9 @@ func unmarshalerEmbeddedStructsHookFunc() mapstructure.DecodeHookFuncValue {
 			// embedded structs passed in via `squash` cannot be pointers. We just check if they are structs:
 			if to.Type().Field(i).IsExported() && to.Type().Field(i).Anonymous {
 				if unmarshaler, ok := to.Field(i).Addr().Interface().(Unmarshaler); ok {
-					if err := unmarshaler.Unmarshal(NewFromStringMap(fromAsMap)); err != nil {
+					c := NewFromStringMap(fromAsMap)
+					c.self = unmarshaler
```

Comment:
Would storing the `skipTopLevelUnmarshaler` bool instead make a difference? If so, we should add a test where it makes a difference.

Reply:
OK, that worked. Until such time there is a reason to revisit, let's keep it simple with your approach.